tools/rna_tools/sortmerna/sortmerna.xml

```diff
@@ -483,6 +483,7 @@ sortmerna --version 2>&1|grep 'SortMeRNA version'
             <output name="aligned_reverse" file="test6_aligned_reverse.fasta" />
             <output name="unaligned_forward" file="test6_unaligned_forward.fasta" />
             <output name="unaligned_reverse" file="test6_unaligned_reverse.fasta" />
+            <output name="output_bam" file="test6_bam.bam" compare="sim_size" delta="200" />
```

Comment:
You can diff BAMs as they are converted to SAM

Reply:
Sorry, I cannot understand.

Reply:
You can remove `compare="sim_size" delta="200"`. Then Galaxy will compare the contents of the bam files (converted to sam).

Reply:
Sorry, but now it is failing with```
 ( /tmp/tmps1odx_h6test3_bam.bam v. /tmp/tmptv8n3jl_test3_bam.bam )
Binary data detected, not displaying diff
```

Reply:
Can you take the generated bam and the test file, convert them to sam and compare them? Probably you just need to set `lines_diff` (as for ordinary text file diff).

Reply:
Yes, you mean in my own computer?

Reply:
Jep, the one in `test-data` and the one generated in the job working dir (`--no_cleanup`).

Reply:
Should I include the SAM as part of the tests? It is bigger than 1M, so not sure how do it.

Reply:
No the conversion was just to check why the test failed.

Reply:
Planemo should show a normal error message with the diff of two files. Is that not what you are seeing?
Galaxy/Planemo simply compares BAM files as they would be simple text files. You could try this with a fake tools

Reply:
Ah, I didn't test it in that way. Let me see.

Reply:
It indicated that both files differed in two lines. I guess now should be fine.

Reply:
Still failing @bgruening. But I did what you mentioned, and it indicated that lines_diff="2".

Reply:
Using `ftype` in tests is a good idea anyway.

Reply:
> Yes, I see `Binary data detected, not displaying diff` this looks wrong to me. Maybe a galaxy bug?
> 
> This here seems to work: https://github.com/galaxyproject/tools-iuc/blob/65fa34f035934b4e9c1fbeb5dffe00b12e6a0f32/tools/crossmap/crossmap_bam.xml#L52
> 
> Can you try adding ftype. Or run the tests in the above example?

Those test passed, but in that case there's not a conversion/ordering by using samtools, so not sure if it could be the reason.

Reply:
It finally worked after adding the ftype tag @bgruening

Reply:
I guess the same trick should also work for Picard. Thanks for trying!